gluecodium/src/test/resources/smoke/generic_types/output/dart/ffi/GenericTypesConversion.cpp

```diff
@@ -0,0 +1,1790 @@
+    return reinterpret_cast<FfiOpaqueHandle>(new (std::nothrow) std::vector<float>::iterator(
+        reinterpret_cast<std::vector<float>*>(handle)->begin()
+    ));
+}
+void
+ListOf_Float_iterator_release_handle(FfiOpaqueHandle iterator_handle) {
+    delete reinterpret_cast<std::vector<float>::iterator*>(iterator_handle);
+}
+bool
+ListOf_Float_iterator_is_valid(FfiOpaqueHandle handle, FfiOpaqueHandle iterator_handle) {
+    return *reinterpret_cast<std::vector<float>::iterator*>(iterator_handle) !=
+        reinterpret_cast<std::vector<float>*>(handle)->end();
+}
+void
+ListOf_Float_iterator_increment(FfiOpaqueHandle iterator_handle) {
+    ++*reinterpret_cast<std::vector<float>::iterator*>(iterator_handle);
+}
+float
+ListOf_Float_iterator_get(FfiOpaqueHandle iterator_handle) {
+    return gluecodium::ffi::Conversion<float>::toFfi(
+        **reinterpret_cast<std::vector<float>::iterator*>(iterator_handle)
+    );
+}
+FfiOpaqueHandle
+ListOf_Int_create_handle() {
+    return reinterpret_cast<FfiOpaqueHandle>(new (std::nothrow) std::vector<int32_t>());
+}
+void
+ListOf_Int_release_handle(FfiOpaqueHandle handle) {
+    if (handle) {
+        delete reinterpret_cast<std::vector<int32_t>*>(handle);
+    }
+}
+void
+ListOf_Int_insert(FfiOpaqueHandle handle, int32_t value) {
+    reinterpret_cast<std::vector<int32_t>*>(handle)->push_back(std::move(
+        gluecodium::ffi::Conversion<int32_t>::toCpp(value)
+    ));
+}
+FfiOpaqueHandle
+ListOf_Int_iterator(FfiOpaqueHandle handle) {
+    return reinterpret_cast<FfiOpaqueHandle>(new (std::nothrow) std::vector<int32_t>::iterator(
+        reinterpret_cast<std::vector<int32_t>*>(handle)->begin()
+    ));
+}
+void
+ListOf_Int_iterator_release_handle(FfiOpaqueHandle iterator_handle) {
+    delete reinterpret_cast<std::vector<int32_t>::iterator*>(iterator_handle);
+}
+bool
+ListOf_Int_iterator_is_valid(FfiOpaqueHandle handle, FfiOpaqueHandle iterator_handle) {
+    return *reinterpret_cast<std::vector<int32_t>::iterator*>(iterator_handle) !=
+        reinterpret_cast<std::vector<int32_t>*>(handle)->end();
+}
+void
+ListOf_Int_iterator_increment(FfiOpaqueHandle iterator_handle) {
+    ++*reinterpret_cast<std::vector<int32_t>::iterator*>(iterator_handle);
+}
+int32_t
+ListOf_Int_iterator_get(FfiOpaqueHandle iterator_handle) {
+    return gluecodium::ffi::Conversion<int32_t>::toFfi(
+        **reinterpret_cast<std::vector<int32_t>::iterator*>(iterator_handle)
+    );
+}
+FfiOpaqueHandle
+ListOf_ListOf_Int_create_handle() {
+    return reinterpret_cast<FfiOpaqueHandle>(new (std::nothrow) std::vector<std::vector<int32_t>>());
+}
+void
+ListOf_ListOf_Int_release_handle(FfiOpaqueHandle handle) {
+    if (handle) {
+        delete reinterpret_cast<std::vector<std::vector<int32_t>>*>(handle);
+    }
+}
+void
+ListOf_ListOf_Int_insert(FfiOpaqueHandle handle, FfiOpaqueHandle value) {
+    reinterpret_cast<std::vector<std::vector<int32_t>>*>(handle)->push_back(std::move(
```

Comment:
I think std::move here is redundant since the return value is a temporary thus rvalue.

Reply:
Done